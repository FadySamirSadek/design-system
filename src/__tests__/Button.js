import React from 'react'
import renderer from 'react-test-renderer'
import { Button, theme } from '..'

describe('Button', () => {
  test('renders', () => {
    const json = renderer.create(<Button />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('custom bg', () => {
    const json = renderer.create(<Button bg="accent" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.accent)
  })

  test('inverted', () => {
    const json = renderer.create(<Button bg="accent" inverted />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.white)
    expect(json).toHaveStyleRule('color', theme.colors.accent)
  })

  test('disabled prop sets', () => {
    const json = renderer.create(<Button disabled />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('opacity', '0.25')
  })

  test('without disabled prop sets', () => {
    const json = renderer.create(<Button bg="success" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.success)
    expect(json).toHaveStyleRule(
      'box-shadow',
      `0 2px 12px 2px ${theme.shadowColor}`,
      {
        modifier: ':hover'
      }
    )
  })
})
