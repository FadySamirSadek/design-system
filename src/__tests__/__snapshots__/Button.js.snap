// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled prop sets 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 4px;
  background-color: #e6f4fc#cce7f8#add9f5#8bc9f0#62b6eb#2d9ee4#298fce#247db4#1d6795#154a6b;
  color: #ffffff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

<button
  className="c0"
  disabled={true}
/>
`;

exports[`Button fullWidth prop sets width to 100% 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 4px;
  background-color: #e6f4fc#cce7f8#add9f5#8bc9f0#62b6eb#2d9ee4#298fce#247db4#1d6795#154a6b;
  color: #ffffff;
  border-width: 0;
  border-style: solid;
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

<button
  className="c0"
/>
`;

exports[`Button renders 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 4px;
  background-color: #e6f4fc#cce7f8#add9f5#8bc9f0#62b6eb#2d9ee4#298fce#247db4#1d6795#154a6b;
  color: #ffffff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

<button
  className="c0"
/>
`;

exports[`Button size large sets height and font-size 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 4px;
  background-color: #e6f4fc#cce7f8#add9f5#8bc9f0#62b6eb#2d9ee4#298fce#247db4#1d6795#154a6b;
  color: #ffffff;
  border-width: 0;
  border-style: solid;
  height: 48px;
  font-size: 16px;
  padding: 0 22px;
}

.c0:disabled {
  opacity: 0.25;
}

<button
  className="c0"
  size="large"
/>
`;

exports[`Button size medium sets height and font-size 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 4px;
  background-color: #e6f4fc#cce7f8#add9f5#8bc9f0#62b6eb#2d9ee4#298fce#247db4#1d6795#154a6b;
  color: #ffffff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

<button
  className="c0"
  size="medium"
/>
`;

exports[`Button size small sets height and font-size 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 4px;
  background-color: #e6f4fc#cce7f8#add9f5#8bc9f0#62b6eb#2d9ee4#298fce#247db4#1d6795#154a6b;
  color: #ffffff;
  border-width: 0;
  border-style: solid;
  height: 32px;
  font-size: 12px;
  padding: 0 12px;
}

.c0:disabled {
  opacity: 0.25;
}

<button
  className="c0"
  size="small"
/>
`;

exports[`Button without disabled prop sets 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 4px;
  background-color: #e6f4fc#cce7f8#add9f5#8bc9f0#62b6eb#2d9ee4#298fce#247db4#1d6795#154a6b;
  color: #ffffff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

<button
  className="c0"
/>
`;
